### PR TITLE
drivers/at24cxxx: merge mtd driver with at24cxxx.c

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

Keeping the `at24cxxx` and it's MTD adaption layer separate just complicates things.
Move the MTD driver into `at24cxxx.c` so we can use internal functions.

The public `at24cxxx_write_page()` never made sense outside the MTD adaption, so drop it.

### CODE REVIEW

File: drivers/at24cxxx/at24cxxx.c [hunk 1]
```diff
@@ -339,3 +321,82 @@ int at24cxxx_disable_write_protect(const at24cxxx_t *dev)
...
+{
+    const at24cxxx_t *dev = DEV(mtd);
+
+    /* some i2c implementations have a limit on the transfer size */
+#ifdef PERIPH_I2C_MAX_BYTES_PER_FRAME
+    size = MIN(size, PERIPH_I2C_MAX_BYTES_PER_FRAME);
+#endif
+
+    i2c_acquire(DEV_I2C_BUS);
+    int res = _read(dev, page * mtd->page_size + offset, dest, size);
```
[the remark this answers is not in the document]
Comment (HendrikVE) [reply drivers/at24cxxx/at24cxxx.c:360]: Ok, so do we want to merge #19258 first then? Apart from this I'm happy. My application is still working with these changes.

File: drivers/at24cxxx/at24cxxx.c [hunk 2]
```diff
@@ -339,3 +321,82 @@ int at24cxxx_disable_write_protect(const at24cxxx_t *dev)
...
+    i2c_release(DEV_I2C_BUS);
+
+    return res < 0 ? res : (int)size;
+}
+
+static int mtd_at24cxxx_write_page(mtd_dev_t *mtd, const void *src, uint32_t page,
```
[the remark this answers is not in the document]
Comment (benpicco) [reply drivers/at24cxxx/at24cxxx.c]: We can also drop the `_at24cxxx` since they are internal functions

**Review by fabian18 — APPROVED** [review]
Filesystem uses MTD and it still works.
```
> 2023-02-16 17:55:00,717 # vfs df
2023-02-16 17:55:00,723 # Mountpoint              Total         Used    Available     Use%
2023-02-16 17:55:00,743 # /nvm0                 32640 B        256 B      32384 B       0%
> vfs ls /nvm0
2023-02-16 17:55:47,890 # vfs ls /nvm0
2023-02-16 17:55:47,897 # ./
2023-02-16 17:55:47,898 # ../
2023-02-16 17:55:47,901 # total 0 files
```

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
bors cancel
bors merge

**Comment by benpicco** [comment]
bors why

**Review by kaspar030 — APPROVED** [review]

**Comment by kaspar030** [comment]
bors merge

**Comment by maribu** [comment]
bors fu

**Comment by fabian18** [comment]
bors merge

**Comment by kaspar030** [comment]
bors merge

**Comment by kaspar030** [comment]
hm. I had already triggered a "synchronize pull requests" in the bors ui.

**Comment by HendrikVE** [comment]
The heck is going on here xD Time to disable E-Mail notifications I guess

**Comment by benpicco** [comment]
bors merge

**Comment by benpicco** [comment]
> The heck is going on here xD Time to disable E-Mail notifications I guess

You are witnessing the first stages of the machine uprising. Bing is rallying online bots behind it's cause - so it begins…

**Review by HendrikVE — APPROVED** [review]

**Comment by HendrikVE** [comment]
I turned my review into "Approve" as well, let's see!?

**Comment by benpicco** [comment]
sudo bors merge

**Comment by benpicco** [comment]
bors merge

**Comment by kaspar030** [comment]
> I turned my review into "Approve" as well, let's see!?

Did you have a started but not approved review?

**Comment by HendrikVE** [comment]
> > I turned my review into "Approve" as well, let's see!?
> 
> Did you have a started but not approved review? _edit_ Or did you just add another review? (trying to figure out what went wrong ...)

Fabian and you had approved it already, while I still had a review open with the "Request changes" option. I thought that shouldn't matter much, as I am not a maintainer and a maintainer's approval should overwrite my review I guessed.

**Comment by benpicco** [comment]
> I still had a review open with the "Request changes" option

Ah I didn't see that

**Pull request merged**